test/watcher.test.js

```diff
@@ -0,0 +1,140 @@
+const path = require('path');
+const ncc = global.coverage ? require("../src/index") : require("../");
+const fs = require('fs');
+
+// Based on the NodeWatchFileSystem class at:
+// - https://github.com/webpack/webpack/blob/master/lib/node/NodeWatchFileSystem.js
+// which in turn exposes:
+// - https://www.npmjs.com/package/watchpack
+
+class CustomWatchFileSystem {
+  constructor(watchStart, watchEnd) {
```

Comment:
`watchStart` and `watchEnd` here are convenience hooks in this test implementation.